xls/build_rules/xls_utilities.bzl

```diff
@@ -306,3 +306,76 @@ Examples:
     implementation = _proto_data_impl,
     attrs = _proto_data_attrs,
 )
+
+def _proto_to_dslx_impl(ctx):
+    """The implementation of the 'proto_to_dslx' rule.
+
+    Args:
+      ctx: The current rule's context object.
+
+    Returns:
+      DefaultInfo provider
+    """
+    schema = ctx.file.proto_schema_path
+    proto2dslx = ctx.executable._proto_to_dslx_tool
+    dslx_filename = get_output_filename_value(
+        ctx,
+        "dslx_file",
+        ctx.attr.name + ".x",
+    )
+    dslx_file = ctx.actions.declare_file(dslx_filename)
+    ctx.actions.run_shell(
+        outputs = [dslx_file],
+        # The files required for generating the protobin file.
+        inputs = [schema, ctx.file.textproto, proto2dslx],
+        # The proto2bin executable is a tool needed by the action.
+        tools = [proto2dslx],
+        command = "{} --proto_def_path {} --proto_name {} --textproto_path {} --var_name {} --output_path {}".format(
+            proto2dslx.path,
+            schema.path,
+            ctx.attr.proto_name,
+            ctx.file.textproto.path,
+            ctx.attr.variable_name,
+            dslx_file.path
+        ),
+        use_default_shell_env = True,
+        mnemonic = "Proto2DSLX",
```

Comment:
curious what the mnemonic is for?

Reply:
This is not required, it is a one-word description of the action